contracts/gnosis_safe_modules/DelayedRecoveryModule.sol

```diff
@@ -0,0 +1,627 @@
+            "Recovery execution failed."
+        );
+
+        delete activeRecoveryInfo;
+
+        emit RecoveryExecuted(
+            _prevOwner,
+            _oldOwner,
+            _newOwner
+        );
+    }
+
+    /**
+     * @notice Aborts the initiated recovery.
+     *
+     * @dev Function requires:
+     *          - There is an initiated recovery with the same tuple of
+     *            addresses (prevOwner, oldOwner, newOwner).
+     *          - Recovery owner has signed execution message.
+     */
+    function abortRecoveryByOwner(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner,
+        bytes32 _r,
+        bytes32 _s,
+        uint8 _v
+    )
+        external
+        activeRecovery
+        validRecovery(_prevOwner, _oldOwner, _newOwner)
+    {
+        bytes32 recoveryHash = hashAbortRecovery(
+            _prevOwner,
+            _oldOwner,
+            _newOwner
+        );
+
+        verify(recoveryHash, _r, _s, _v);
+
+        delete activeRecoveryInfo;
+
+        emit RecoveryAborted(
+            _prevOwner,
+            _oldOwner,
+            _newOwner
+        );
+    }
+
+    /**
+     * @notice Aborts the initiated recovery.
+     *
+     * @dev Function requires:
+     *          - Only recovery controller can call.
+     *          - There is an initiated recovery with the same tuple of
+     *            addresses (prevOwner, oldOwner, newOwner).
+     */
+    function abortRecoveryByController(
+        address _prevOwner,
+        address _oldOwner,
+        address _newOwner
+    )
+        external
+        onlyRecoveryController
+        activeRecovery
+        validRecovery(_prevOwner, _oldOwner, _newOwner)
+    {
+        delete activeRecoveryInfo;
+
+        emit RecoveryAborted(
+            _prevOwner,
+            _oldOwner,
+            _newOwner
+        );
+    }
+
+    function resetRecoveryOwner(
+        address _newRecoveryOwner,
+        bytes32 _r,
+        bytes32 _s,
+        uint8 _v
+    )
+        external
+        onlyRecoveryController
+    {
+        require(
+            _newRecoveryOwner != address(0),
+            "New recovery owner's address is null."
+        );
+
+        bytes32 resetRecoveryOwnerHash = hashResetRecoveryOwner(
+            recoveryOwner,
+            _newRecoveryOwner
+        );
+
+        verify(resetRecoveryOwnerHash, _r, _s, _v);
+
+        address oldRecoveryOwner = recoveryOwner;
+
+        recoveryOwner = _newRecoveryOwner;
+
+        emit ResetRecoveryOwner(
+            oldRecoveryOwner,
+            recoveryOwner
+        );
+    }
+
+
+    /* Private Functions */
+
+    function verify(
+        bytes32 _digest,
+        bytes32 _r,
+        bytes32 _s,
+        uint8 _v
+    )
+        private
+        view
+    {
+        require(
+            ecrecover(_digest, _v, _r, _s) == recoveryOwner,
+            "The recovery owner does not sign the message."
+        );
+    }
+
+    function hashRecovery(
```

Comment:
please collapse the private/view functions as discussed

Reply:
Done.